packages/wallet/backend/src/account/controller.ts

```diff
@@ -51,15 +50,19 @@ export class AccountController implements IAccountController {
     next: NextFunction
   ) => {
     const userId = req.session.user.id
+    const shouldGraphFetched = req.query['include'] == 'paymentPointers'
 
     try {
-      const getAccountsResult = await this.deps.accountService.getAccounts(
-        userId
-      )
+      let accounts
+      if (shouldGraphFetched)
```

Comment:
I think you should pass 'paymentPointers' as optional args to the already existing getAccounts method, in order to query with graph fetched instead of adding the 'getAccountsWithGraphFetched' method

Reply:
yeap, I agree with this one @

Reply:
Well, in that case getAccounts function will get complex again, and it's same as previous version. But @dragosp1011 and @raducristianpopa wanted it to be separated...

Reply:
I think something like this should work
```
const user = await User.query().findById(userId)

    if (!user || !user.rapydWalletId) {
      throw new NotFound()
    }

    const query = Account.query().where('userId', userId)

    if (paymentPointers) {
      query.withGraphFetched('paymentPointers')
    }

    const accounts = await query

    if (balance) {
      const accountsBalance = await this.deps.rapyd.getAccountsBalance(
          user.rapydWalletId
      )

      accounts.forEach((acc) => {
        acc.balance = transformBalance(
            accountsBalance.data?.find(
                (rapydAccount) => rapydAccount.currency === acc.assetCode
            )?.balance ?? 0,
            acc.assetScale
        )
      })
    }

    return accounts
    ```

Reply:
@raducristianpopa what do you think? is this what you had in mind?